src/elements/NativeComponents.tsx

```diff
@@ -1,31 +1,29 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';
 
-declare module 'react-native' {
-  function requireNativeComponent(
-    viewName: string,
-  ): // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  React.ComponentType<React.PropsWithChildren<any>>;
-}
-
-export const RNSVGSvg = rnc('RNSVGSvgView');
-export const RNSVGCircle = rnc('RNSVGCircle');
-export const RNSVGClipPath = rnc('RNSVGClipPath');
-export const RNSVGDefs = rnc('RNSVGDefs');
-export const RNSVGEllipse = rnc('RNSVGEllipse');
-export const RNSVGForeignObject = rnc('RNSVGForeignObject');
-export const RNSVGGroup = rnc('RNSVGGroup');
-export const RNSVGImage = rnc('RNSVGImage');
-export const RNSVGLine = rnc('RNSVGLine');
-export const RNSVGLinearGradient = rnc('RNSVGLinearGradient');
-export const RNSVGMarker = rnc('RNSVGMarker');
-export const RNSVGMask = rnc('RNSVGMask');
-export const RNSVGPath = rnc('RNSVGPath');
-export const RNSVGPattern = rnc('RNSVGPattern');
-export const RNSVGRadialGradient = rnc('RNSVGRadialGradient');
-export const RNSVGRect = rnc('RNSVGRect');
-export const RNSVGSymbol = rnc('RNSVGSymbol');
-export const RNSVGText = rnc('RNSVGText');
-export const RNSVGTextPath = rnc('RNSVGTextPath');
-export const RNSVGTSpan = rnc('RNSVGTSpan');
-export const RNSVGUse = rnc('RNSVGUse');
+export const RNSVGSvg = codegenNativeComponent<any>('RNSVGSvgView');
+export const RNSVGCircle = codegenNativeComponent<any>('RNSVGCircle');
+export const RNSVGClipPath = codegenNativeComponent<any>('RNSVGClipPath');
+export const RNSVGDefs = codegenNativeComponent<any>('RNSVGDefs');
+export const RNSVGEllipse = codegenNativeComponent<any>('RNSVGEllipse');
+export const RNSVGForeignObject =
+  codegenNativeComponent<any>('RNSVGForeignObject');
+export const RNSVGGroup = codegenNativeComponent<any>('RNSVGGroup');
+export const RNSVGImage = codegenNativeComponent<any>('RNSVGImage');
+export const RNSVGLine = codegenNativeComponent<any>('RNSVGLine');
+export const RNSVGLinearGradient = codegenNativeComponent<any>(
+  'RNSVGLinearGradient',
+);
+export const RNSVGMarker = codegenNativeComponent<any>('RNSVGMarker');
+export const RNSVGMask = codegenNativeComponent<any>('RNSVGMask');
+export const RNSVGPath = codegenNativeComponent<any>('RNSVGPath');
+export const RNSVGPattern = codegenNativeComponent<any>('RNSVGPattern');
+export const RNSVGRadialGradient = codegenNativeComponent<any>(
+  'RNSVGRadialGradient',
+);
+export const RNSVGRect = codegenNativeComponent<any>('RNSVGRect');
+export const RNSVGSymbol = codegenNativeComponent<any>('RNSVGSymbol');
+export const RNSVGText = codegenNativeComponent<any>('RNSVGText');
+export const RNSVGTextPath = codegenNativeComponent<any>('RNSVGTextPath');
+export const RNSVGTSpan = codegenNativeComponent<any>('RNSVGTSpan');
+export const RNSVGUse = codegenNativeComponent<any>('RNSVGUse');
```

Comment:
@WoLewicki, this won't work unfortunately.

The type passed into codegenNativeComponent needs to be declared in the same file as the call to codegenNativeComponent. Since we're passing any to all these codegenNativeComponent calls, the babel plugin will not transform these into NativeComponetRegistry.get calls. 

So, ideally, the setup should be like this:
1. We delete this NativeComponents.ts file.
2. All TypeScript files use the spec files defined in src/fabric

We might have to translate the files inside src/fabric into typescript.